components/brave_vpn/brave_vpn_service_desktop.cc

```diff
@@ -61,12 +67,11 @@ bool IsValidRegion(const brave_vpn::mojom::Region& region) {
 }
 
 std::string GetBraveVPNPaymentsEnv() {
-  const std::string env = brave_rewards::GetEnvironment();
-  if (env == brave_rewards::kEnvProduction)
+  const std::string env = skus::GetEnvironment();
```

Comment:
why does vpn env get set by the skus environment and why is it `""` for production?

Reply:
On desktop, the environment is tied to SKUs because you buy it from account.brave.com (or similar). The credentials for VPN will always be together with the SKU environment. When the vendor (Guardian) receives a credential, it also has an environment. They then can do a lookup using Payment Service (which @evq provided an implementation for)

Reply:
this needs comments

Reply:
Comments added in https://github.com/brave/brave-core/pull/10847/commits/f1fd81200a1006b5c0971b6805968acf4934c79a